Corrfunc/tests/test_theory.py

```diff
@@ -32,7 +33,223 @@ def test_DD(gals_Mr19, isa, nthreads):
+    periodic_ref = pjoin(dirname(abspath(__file__)),
+                         "../../theory/tests/",
+                         "Mr19_{}_periodic".format(funcname))
+    nonperiodic_ref = pjoin(dirname(abspath(__file__)),
+                            "../../theory/tests/",
+                            "Mr19_{}_nonperiodic".format(funcname))
+    autocorr = 1
+    periodic = 1
+    pimax = 40.0
+    mu_max = 0.5
+    nmu_bins = 10
+
+    x, y, z, w = gals_Mr19
+    args = [autocorr, nthreads, binfile, x, y, z]
+    kwargs = dict(weights1=w, weight_type='pair_product',
+                  periodic=periodic,
+                  verbose=True, isa=isa)
+
+    if funcname == 'DDrppi':
+        args.insert(2, pimax)
+        kwargs['output_rpavg'] = True
+        ravg_name = 'rpavg'
+    elif funcname == 'DDsmu':
+        kwargs['output_savg'] = True
+        args[3:3] = (mu_max, nmu_bins)
+        ravg_name = 'savg'
+    else:
+        kwargs['output_ravg'] = True
+        ravg_name = 'ravg'
+
+    # scalar periodic
+    results_DD = func(*args, boxsize=boxsize, **kwargs)
+    check_against_reference(results_DD, periodic_ref,
+                            ravg_name=ravg_name, ref_cols=(0, 4, 1))
+
+    # 0-d array, periodic
+    results_DD = func(*args, boxsize=np.array(boxsize), **kwargs)
+    check_against_reference(results_DD, periodic_ref,
+                            ravg_name=ravg_name, ref_cols=(0, 4, 1))
+
+    # 1-d array, len 1, periodic
+    results_DD = func(*args, boxsize=np.array([boxsize]), **kwargs)
+    check_against_reference(results_DD, periodic_ref,
+                            ravg_name=ravg_name, ref_cols=(0, 4, 1))
+
+    # 1-d array, len 3, periodic
+    results_DD = func(*args, boxsize=np.array([boxsize, boxsize, boxsize]),
+                      **kwargs)
+    check_against_reference(results_DD, periodic_ref,
+                            ravg_name=ravg_name, ref_cols=(0, 4, 1))
+
+    # 3-tuple periodic cube
+    results_DD = func(*args, boxsize=(boxsize, boxsize, boxsize), **kwargs)
+    check_against_reference(results_DD, periodic_ref,
+                            ravg_name=ravg_name, ref_cols=(0, 4, 1))
+
+    # scalar non-periodic
+    results_DD = func(*args, boxsize=(-1., -1., -1.), **kwargs)
+    check_against_reference(results_DD, nonperiodic_ref,
+                            ravg_name=ravg_name, ref_cols=(0, 4, 1))
+
+    # non-periodic via oversized box
+    results_DD = func(*args, boxsize=(2000., 2000., 2000.), **kwargs)
+    check_against_reference(results_DD, nonperiodic_ref,
+                            ravg_name=ravg_name, ref_cols=(0, 4, 1))
+
+
+@pytest.mark.parametrize('N', [1, 2])
+def test_narrow_extent(N, isa='fastest', nthreads=maxthreads()):
+    '''Test that a narrow particle distribution in a large box
+    does not throw an error.
+    Regression test for
+    https://github.com/manodeep/Corrfunc/pull/276#issuecomment-1164872508
+    '''
+    from Corrfunc.theory import DD
```

Comment:
Is it possible to extend the tests to DDrppi and DDsmu?

Reply:
That's true, we did modify those code paths so we should test them.  However, I think the right place is in the brute-force test below; this particular test is a gridlink regression test, unrelated to the pair counter.  I've added DDrppi and DDsmu to the brute-force test.